client/src/components/RatingTile.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+
+const RatingTile = ({ id, content, activityId }) => {
+  const [ratings, setRatings] = useState({});
+  const [allRatings, setAllRatings] = useState([]);
+
+  const postRating = async (reviewId, selectedRating) => {
+    try {
+      const response = await fetch(`/api/v1/activities/${activityId}/reviews/rating`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ reviewId, rating: selectedRating }),
+      });
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`);
+      }
+      const responseData = await response.json();
+      setRatings(responseData);
+    } catch (error) {
+      console.error("Error in fetch!", error.message);
+    }
+  };
+
+  let upVotes = 0;
+  let downVotes = 0;
+
+  allRatings.forEach((ratingObject) => {
+    if (id === ratingObject.reviewId && ratingObject.rating === 2) {
+      upVotes++;
+    }
+    if (id === ratingObject.reviewId && ratingObject.rating === 1) {
+      downVotes++;
+    }
+  });
```

Comment:
you can compute this on the backend, so that we dont need to do it here! 

add this computation to your serializer!!!